pkg/packetbrokeragent/grpc_gspba.go

```diff
@@ -157,16 +166,44 @@ func (s *gsPbaServer) UpdateGateway(ctx context.Context, req *ttnpb.UpdatePacket
 		}
 	}
 
+	var adminContact, techContact *packetbroker.ContactInfo
 	if ttnpb.HasAnyField(req.FieldMask.GetPaths(), "contact_info") {
-		adminContact, techContact := toPBContactInfo(req.Gateway.GetContactInfo())
-		updateReq.AdministrativeContact = &packetbroker.ContactInfoValue{
-			Value: adminContact,
+		adminContact, techContact = toPBContactInfo(req.Gateway.GetContactInfo()) // nolint: staticcheck
+	}
+
+	if ttnpb.HasAnyField(req.FieldMask.GetPaths(), "administrative_contact") {
+		// Only replaces the administrative contact if no error happens.
+		adminUsr, err := s.fetchUserFromContact(
+			ctx, req.Gateway.GetAdministrativeContact(),
+			func(o *ttnpb.Organization) *ttnpb.OrganizationOrUserIdentifiers { return o.GetAdministrativeContact() },
+		)
+		if err == nil {
+			adminContact = &packetbroker.ContactInfo{Name: adminUsr.Ids.UserId, Email: adminUsr.PrimaryEmailAddress}
+		} else if !errors.IsNotFound(err) {
+			logger.WithField("gatewayID", req.Gateway.Ids).WithError(err).Warn("Failed to fetch administrative contact")
```

Comment:
`snake_case` for log fields. I also don't think we should log this error for cases in which the email is not validated.